src/Server/Server.fs

```diff
@@ -1091,33 +1079,34 @@ let listResourcesIfInTemplate
         return empty
 }
 
-let getRemappedImportProps 
+let getRemappedImportProps
     (resource: AwsCloudFormationResource)
     (resourceData: Dictionary<string, Dictionary<string,string>>)
     (resourceContext: AwsResourceContext)
-    : Option<RemappedSpecResult> =      
-    
-    AwsCloudFormationTemplates.remapSpecifications
-    |> Seq.tryFind (fun pair -> pair.Key = resource.resourceType)
-    |> Option.map (fun pair -> pair.Value)
-    |> Option.filter (fun spec -> spec.validatorFunc resource resourceData spec)
-    |> Option.map (fun spec -> spec.remapFunc resource resourceData resourceContext spec)
+    : Result<RemappedSpecResult, string> =      
 
-let getRemappedImportPropsV2
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (resourceContext: AwsResourceContext)
-    : Option<RemappedSpecResult> =      
-    
-    AwsCloudFormationTemplates.remapSpecificationsV2
+    AwsCloudFormationTemplates.importIdentityBuilders
     |> Map.tryFind resource.resourceType
-    |> Option.filter (fun _ -> resourceData.ContainsKey resource.logicalId)
-    |> Option.bind (fun remapFunc -> 
-        let data = resourceData[resource.logicalId]
-        match remapFunc resource data resourceContext with
-        | Ok remappedSpecResult -> Some remappedSpecResult
-        | Error remapError -> None // todo: log error or return it to user
-    ) 
+    |> function 
+        | None ->
+            Error $"Found no mapping specification for resource type {resource.resourceType}"
+        | Some _ when not (resourceData.ContainsKey resource.logicalId) -> 
+            Error $"Resource '{resource.logicalId}' of type {resource.resourceType} has no data"
```

Comment:
`has no data` is a little vague - its really that the data is not in the intermediate layer for some reason - might be useful to make this error more specific for debugging purposes (not that I expect that we would see this case often)